File: src/main/java/io/github/mschout/gitlab/toggltimer/configuration/GitlabConfiguration.java
package io.github.mschout.gitlab.toggltimer.configuration;

import static java.util.Objects.requireNonNull;

import org.gitlab4j.api.GitLabApi;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class GitlabConfiguration {

  @Value("${gitlab.accessToken}")
  private String accessToken;

  @Value("${gitlab.url:https://gitlab.com}")
  private String gitlabUrl;

  @Bean
  public GitLabApi gitlabApi() {
    requireNonNull(accessToken, "accessToken is required");
    return new GitLabApi(gitlabUrl, accessToken);
  }
}

File: src/main/kotlin/io/github/mschout/gitlab/toggltimer/configuration/GitlabConfiguration.kt
package io.github.mschout.gitlab.toggltimer.configuration

import org.gitlab4j.api.GitLabApi
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class GitlabConfiguration(
    @param:Value("\${gitlab.accessToken}") private val accessToken: String,
    @param:Value("\${gitlab.url:https://gitlab.com}") private val gitlabUrl: String,
) {
  @Bean fun gitlabApi(): GitLabApi = GitLabApi(gitlabUrl, accessToken)
}
